Source/Renderer/VertexArray.js

```diff
@@ -636,9 +673,52 @@ define([
         return this._attributes[index];
     };
 
+    // Work around for ANGLE, where the attribute divisor seems to be part of the global state instead
+    // of the VAO state. This function is called when the vao is bound, and should be removed
+    // once the ANGLE issue is resolved.
+    var divisors = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]; // at most 16 vertex attributes
+    var hadInstancedAttributes = false;
+
+    function setVertexAttribDivisor(vertexArray) {
+        var hasInstancedAttributes = vertexArray._hasInstancedAttributes;
+        if (!hasInstancedAttributes && !hadInstancedAttributes) {
+            return;
+        }
+        hadInstancedAttributes = hasInstancedAttributes;
+
+        var attributes = vertexArray._attributes;
+        var maxAttributes = ContextLimits.maximumVertexAttributes;
+        var instancedArraysExtension = vertexArray._instancedArraysExtension;
+        var i;
+
+        if (hasInstancedAttributes) {
+            var length = attributes.length;
+            for (i = 0; i < length; ++i) {
+                var attribute = attributes[i];
+                if (attribute.enabled) {
+                    var divisor = attribute.instanceDivisor;
+                    var index = attribute.index;
+                    if (divisor !== divisors[index]) {
+                        instancedArraysExtension.vertexAttribDivisorANGLE(index, divisor);
+                        divisors[index] = divisor;
+                    }
+                }
+            }
+        } else {
+            for (i = 0; i < maxAttributes; ++i) {
```

Comment:
We can do better than `maxAttributes`, right?  Why not track the max attribute index used for the previous draw call and just loop to that?  It's still not optimal, but most likely pretty good.